cmd/operator-sdk/generate/bundle/cmd.go

```diff
@@ -0,0 +1,123 @@
+
+	kbutil "github.com/operator-framework/operator-sdk/internal/util/kubebuilder"
+)
+
+//nolint:maligned
+type bundleCmd struct {
+	// Options to turn on different parts of bundling.
+	kustomize bool
+	manifests bool
+	metadata  bool
+
+	// Common options.
+	operatorName string
+	version      string
+	inputDir     string
+	outputDir    string
+	manifestRoot string
+	apisDir      string
+	crdsDir      string
+	stdout       bool
+	quiet        bool
+
+	// Metadata options.
+	channels       string
+	defaultChannel string
+	overwrite      bool
+}
+
+//nolint:lll
+func NewCmd() *cobra.Command {
+	c := &bundleCmd{}
+	cmd := &cobra.Command{
+		Use:   "bundle",
+		Short: "Generates bundle data for the operator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
+			}
+
+			// Turn them all on by default here to simplify individual flag usage.
+			fs := cmd.Flags()
+			if !fs.Changed("kustomize") && !fs.Changed("metadata") && !fs.Changed("manifests") {
+				c.kustomize = true
+				c.manifests = true
+				c.metadata = true
+			}
+
+			cfg, err := kbutil.ReadConfig()
+			if err != nil {
+				return fmt.Errorf("error reading configuration: %v", err)
+			}
+			c.setDefaults(cfg)
+
+			if c.kustomize {
+				if err = c.runKustomize(cfg); err != nil {
+					log.Fatalf("Error generating bundle bases: %v", err)
+				}
+			}
+			if c.manifests {
+				if err = c.validateManifests(cfg); err != nil {
+					return fmt.Errorf("invalid command options: %v", err)
+				}
+				if err = c.runManifests(cfg); err != nil {
+					log.Fatalf("Error generating bundle manifests: %v", err)
+				}
+			}
+			if c.metadata {
+				if err = c.runMetadata(); err != nil {
+					log.Fatalf("Error generating bundle metadata: %v", err)
+				}
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().BoolVar(&c.kustomize, "kustomize", false, "Generate kustomize bases")
+	cmd.Flags().BoolVar(&c.manifests, "manifests", false, "Generate bundle manifests")
+	cmd.Flags().BoolVar(&c.metadata, "metadata", false, "Generate bundle metadata and Dockerfile")
+	cmd.Flags().BoolVar(&c.stdout, "stdout", false, "Write bundle manifest to stdout")
+
+	c.addCommonFlagsTo(cmd.Flags())
+
+	return cmd
+}
+
+// TODO(estroz): add flag to skip API metadata regeneration.
+func (c *bundleCmd) addCommonFlagsTo(fs *pflag.FlagSet) {
+	fs.StringVar(&c.operatorName, "operator-name", "", "Name of the bundle's operator")
+	fs.StringVarP(&c.version, "version", "v", "", "Semantic version of the operator in the generated bundle. "+
+		"Only set if creating a new bundle or upgrading your operator")
+	fs.StringVar(&c.inputDir, "input-dir", "", "Directory to read an existing bundle format from")
+	fs.StringVar(&c.outputDir, "output-dir", "", "Directory to write the bundle format to")
+	fs.StringVar(&c.manifestRoot, "manifest-root", "", "Root directory for operator manifests, ex. Deployment and RBAC")
```

Comment:
@estroz So this is effectively `--deploy-dir` right?
I guess this is subjective but `manifests` immediately made me think of the bundle manifests directory. 
I wonder if a better name wouldn't be `operator-manifests-dir` similar to `crds-dir` even if it's more verbose.
Also correct me if I'm wrong but this dir is used for Custom Resource manifests collection, and not the crd dir as mentioned below:
```suggestion
	fs.StringVar(&c.manifestRoot, "manifest-root", "", "Root directory for operator manifests: Deployment, RBAC and CustomResources.")
```

Reply:
`operator-manifests-dir` doesn't seem different enough to me. Perhaps a better solution is to be explicit about the difference between `--input-dir` and `--manifest-root` in flag help.